### PR TITLE
[IMP] *: Partner form improvement

### PR DESCRIPTION
**Pull request by malb-odoo**
This commit will do a lot of change on the partner form:
- Removing res.partner.title model and field
- Removing/modifying some help on fields
- Making the bank_ids list view into an editable tag that has 2 color depending on if the bank is verified or not
- Add new widget many2many_tags_banks for the banks tags
- Remove some groups and modify existing one
- Change in some placement of smart button on top of the form
- Change in the child_ids of the partner form (naming of the horizontal radio
button, remove input)
- Add properties on the partner form
- Change the image to fit with v18 ux change
- Warning potential duplicate change
- Hide the country in the cart of the address in the partner form

task: 4306429

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/static/src/components/many2many_tags_banks/many2many_tags_banks.js
Comment (SaddemAmine) [inline addons/account/static/src/components/many2many_tags_banks/many2many_tags_banks.js:9]: I'm not sure but if this class necessary ? Can't we use `TagList` directly ?

File: addons/im_livechat/tests/test_js.py
Comment (malb-odoo) [inline addons/im_livechat/tests/test_js.py:17]: forgot to remove that, i will wait for the review to be done to remove it

**Review by SaddemAmine — APPROVED** [review]
LGTM except for a tiny typo in your commit message (cart) and a failing JS test 👀

File: addons/account/models/res_partner_bank.py
Comment (FlorianGilbert) [inline addons/account/models/res_partner_bank.py:51]:
```suggestion
    color = fields.Integer(compute='_compute_color')
```
Not necessary

**Review by FlorianGilbert — CHANGES_REQUESTED** [review]
In your commit message, you refer to the product form view, isn't it the partner one?

Could you please confirm me that localization changes are not in the spec? I can't find anything about it.
I didn't see the change for the industry field and for the add button in kanban view, is that normal?

File: odoo/addons/base/i18n/base.pot
Comment (malb-odoo) [reply odoo/addons/base/i18n/base.pot:27837]: I will export the pot and po again, i did the change because we had a test that was failing since we check that translation exactly

File: odoo/addons/base/i18n/base.pot
Comment (malb-odoo) [reply odoo/addons/base/i18n/base.pot:1]: If i'm not mistaken in master no need to export pot and po, but in this case, a test was checking if the term other address was well translated and so i had to change that

File: odoo/addons/base/views/res_partner_views.xml
Comment (malb-odoo) [reply odoo/addons/base/views/res_partner_views.xml:182]: If i remember correctly, if we put the properties field in a group it wasn't displayed but i will check again

File: odoo/addons/base/views/res_partner_views.xml
Comment (malb-odoo) [reply odoo/addons/base/views/res_partner_views.xml:199]: image
PS: the two last point will be done by the JS team

File: addons/purchase/views/res_partner_views.xml
Comment (malb-odoo) [reply addons/purchase/views/res_partner_views.xml:16]: The rest seems to be working fine, but the maxLength doesn't work tough

**Review by tde-banana-odoo — COMMENTED** [review]
SM part review. Thanks for the cleaning and title removal, it simplifies things :) 

Have to say however the properties is probably something that should be extracted and thought again as it feels quite wrong to me.

Cheers !

File: odoo/addons/base/views/res_partner_views.xml
Comment (FlorianGilbert) [reply odoo/addons/base/views/res_partner_views.xml:257]: @Olivier-olma

**Review by FlorianGilbert — APPROVED** [review]
Except the case for the `company_regitry` instead of the `company_id`
LGTM

File: addons/web/static/tests/_framework/mock_server/mock_model.js
Comment (tde-banana-odoo) [inline addons/web/static/tests/_framework/mock_server/mock_model.js:2990]: Is it still relevant ? What does it fix ?

**Comment by malb-odoo** [comment]
@tde-banana-odoo Thanks for the reviews and help ! Could you take a last look before we merge this :smile:

**Comment by FlorianGilbert** [comment]
@robodoo r+

**Comment by QaidjoharBarbhaya** [comment]
Keeping title field might have helped in identifying gender of contact.

**Comment by QaidjoharBarbhaya** [comment]
@malb-odoo

**Pull request closed without merging**
